mbdynAdapter/adapter.py

```diff
@@ -1,81 +1,181 @@
-import numpy as np
-from mpi4py import MPI
-import precice
-
-class MBDynAdapter:
-    def __init__(self, mbdHelper, configFileName = "../precice-config.xml"):
-        self.mbd = mbdHelper
-        self.interface = precice.Interface("Structure_Solver", 0, 1) # proc no, nprocs
-        self.interface.configure(configFileName)
-        self.dim = self.interface.get_dimensions()
-        nodes = self.mbd.getNodes()
-
-        # if self.dim == 2:
-            # self.nodesid = np.where(nodes[:,2] < 1e-6)
-            # nodes = nodes[self.nodesid]
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
 
-        self.nnodes = len(nodes)
+import logging
 
-        nmeshID = self.interface.get_mesh_id("Structure_Nodes")
-        self.nodeVertexIDs = self.nnodes*[0.0]
-        self.interface.set_mesh_vertices(nmeshID, self.nnodes, np.ravel(nodes[:,:self.dim]), self.nodeVertexIDs)
-        self.displacements = np.array(self.dim*self.nnodes*[0.0])
+import numpy as np
+import precice
 
-        # ccs = self.mbd.getCellCenters()
-        # self.ncells = len(ccs)
-        # cmeshID = self.interface.get_mesh_id("Structure_CellCenters")
-        # self.cellVertexIDs = self.ncells*[0.0]
-        # self.interface.set_mesh_vertices(cmeshID, self.ncells, np.ravel(ccs[:,:self.dim]), self.cellVertexIDs)
-        self.force = np.array(self.dim*self.nnodes*[0.0])
+from helper import MBDynHelper, PreciceHelper
+from input import MBDynInput
 
-        self.displacementsID = self.interface.get_data_id("DisplacementDelta", nmeshID)
-        self.forceID = self.interface.get_data_id("Force", nmeshID)
+# create logger
+module_logger = logging.getLogger('adapter')
 
-        self.dt = self.interface.initialize()
-        self.mbd.controlDict["timeStep"] = self.dt
-        self.mbd.initializeMBDyn()
 
+class MBDynAdapter:
+    def __init__(self, mbdyn_prep, config_file_name='../precice-config.xml',
```

Comment:
Is there a better way to pass the preCICE config file to the adapter? This is a hard coding which is not very user-friendly.